eng/ApiCompat/ApiCompat.targets

```diff
@@ -0,0 +1,131 @@
+                              Condition="'%(FrameworkListFileClass.Profile)' == 'WPF'" />
+    </ItemGroup>
+
+    <!-- Validate paths for the collected items -->
+    <Error Text="One ofthe required contract dependencies directories is invalid."
+           Condition="!Exists('%(_ContractDependencyDirectories.FullPath)')"/>
+    <Error Text="One ofthe required implementation directories is invalid."
+           Condition="!Exists('%(_DependencyDirectories.FullPath)')"/>
+    <Error Text="Some of Windows Forms assemblies referenced in 'FrameworkListFileClass' group can't be found."
+           Condition="!Exists('%(_WindowsFormsFilePaths.FullPath)')"/>
+
+    <PropertyGroup>
+      <_WindowsFormsApiCompatArgs>"@(_WindowsFormsFilePaths, ',')"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --contract-depends "$(OutputPath),@(_ContractDependencyDirectories, ','),"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --impl-dirs "@(_DependencyDirectories, ','),"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs Condition="'$(CreateBaseline)' != 'true'">$(_WindowsFormsApiCompatArgs) --baseline "$(_WindowsFormsApiCompatBaselineIgnoreFile)"</_WindowsFormsApiCompatArgs>
+      <_WindowsFormsApiCompatArgs>$(_WindowsFormsApiCompatArgs) --ignore-design-time-facades</_WindowsFormsApiCompatArgs>
+
+      <_WindowsFormsCreateBaselineArgs Condition="'$(CreateBaseline)' == 'true'"> &gt; "$(_WindowsFormsApiCompatBaselineIgnoreFile)"</_WindowsFormsCreateBaselineArgs>
+
+      <_WindowsFormsApiCompatResponseFile>$(IntermediateOutputPath)apicompat.winforms.rsp</_WindowsFormsApiCompatResponseFile>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <_WpfApiCompatArgs>"@(_WpfFilePaths, ',')"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs>$(_WpfApiCompatArgs) --contract-depends "$(OutputPath),@(_ContractDependencyDirectories, ','),"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs>$(_WpfApiCompatArgs) --impl-dirs "@(_DependencyDirectories, ','),"</_WpfApiCompatArgs>
+      <_WpfApiCompatArgs Condition="'$(CreateBaseline)' != 'true'">$(_WindowsFormsApiCompatArgs) --baseline "$(_WpfApiCompatBaselineIgnoreFile)"</_WpfApiCompatArgs>
+      <_WpfCreateBaselineArgs Condition="'$(CreateBaseline)' == 'true'"> &gt; "$(_WpfApiCompatBaselineIgnoreFile)"</_WpfCreateBaselineArgs>
+
+      <_WpfApiCompatResponseFile>$(IntermediateOutputPath)apicompat.wpf.rsp</_WpfApiCompatResponseFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(IntermediateOutputPath)" />
+    <WriteLinesToFile File="$(_WindowsFormsApiCompatResponseFile)" Lines="$(_WindowsFormsApiCompatArgs)" Overwrite="true" />
+    <WriteLinesToFile File="$(_WpfApiCompatResponseFile)" Lines="$(_WpfApiCompatArgs)" Overwrite="true" />
+  </Target>
+
+  <!--
+    Validate that the current servicing release retains the same public API surface as the first official release.
```

Comment:
This looks good for servicing, some notes for later: normally you want to run this in the opposite direction during regular development.  That helps you make sure you don't break compat over last release.  It could be interesting to consider that when writing these targets and support both modes.  Perhaps some feedback for when this goes to `main`